src/text/line.rs

```diff
@@ -453,6 +453,39 @@ impl<'a> Line<'a> {
         self.spans.iter_mut()
     }
 
+    /// Returns a line that's truncated corresponding to it's alignment and result width
+    #[must_use = "method returns the modified value"]
+    pub fn truncated(&'a self, result_width: u16) -> Self {
```

Comment:
I think I'd prefer this to be private for now. The feature is useful, but the name of this method likely shouldn't be `truncated`. It is more similar to the `split_at` (and other related) functions on str, than to the `truncate` method (which implies deletion of characters from a mutable value).